### PR TITLE
update pipelining tutorial

### PR DESCRIPTION
**Pull request by H-Huang**
Update the tutorial to work with latest API changes

### CODE REVIEW

File: intermediate_source/pipelining_tutorial.rst [hunk 1]
```diff
@@ -232,5 +231,10 @@ We discussed two methods of model partitioning, manual and tracer-based, and dem
 micro-batches across different stages. Finally, we covered the execution of the pipeline schedule and the launch of distributed
 processes using ``torchrun``.
 
-For a production ready usage of pipeline parallelism as well as composition with other distributed techniques, see also
+Additional Resources
+--------------------
+
+We have successfully integrated torch.distributed.pipelining into the torchtitan repository. TorchTitan is a clean, minimal code base for
```
Comment (svekars) [inline intermediate_source/pipelining_tutorial.rst:237]: Link to to torchtiton repo?
```suggestion
We have successfully integrated ``torch.distributed.pipelining`` into the `torchtitan repository <https://github.com/pytorch/torchtitan>`__. TorchTitan is a clean, minimal code base for
```

File: intermediate_source/pipelining_tutorial.rst [hunk 2]
```diff
@@ -232,5 +231,10 @@ We discussed two methods of model partitioning, manual and tracer-based, and dem
...
+Large-scale LLM training using native PyTorch. For a production ready usage of pipeline
```
Comment (svekars) [inline intermediate_source/pipelining_tutorial.rst:238]:
```suggestion
large-scale LLM training using native PyTorch. For a production ready usage of pipeline
```

File: intermediate_source/pipelining_tutorial.rst [hunk 3]
```diff
@@ -109,32 +109,29 @@ Step 1: Partition the Transformer Model
 There are two different ways of partitioning the model:
 
 First is the manual mode in which we can manually create two instances of the model by deleting portions of
-attributes of the model. In this example for a 2 stage (2 ranks) the model is cut in half.
+attributes of the model. In this example for 2 stages (2 ranks) the model is cut in half.
```
Comment (svekars) [inline intermediate_source/pipelining_tutorial.rst:112]:
```suggestion
attributes of the model. In this example for two stages (2 ranks), the model is cut in half.
```

**Review by svekars — APPROVED** [review]
LGTM! Minor editorial nits

File: intermediate_source/pipelining_tutorial.rst [hunk 4]
```diff
@@ -181,18 +178,19 @@ as well as multiple-stage-per-rank schedules such as ``Interleaved1F1B`` and ``L
...
 
+      model.to(device)
       x = x.to(device)
       y = y.to(device)
 
       def tokenwise_loss_fn(outputs, targets):
          loss_fn = nn.CrossEntropyLoss()
-         outputs = outputs.view(-1, model_args.vocab_size)
-         targets = targets.view(-1)
+         outputs = outputs.reshape(-1, model_args.vocab_size)
```
Comment (wconstab) [inline intermediate_source/pipelining_tutorial.rst:192]: Just curious, why this change?

**Review by wconstab — APPROVED** [review]
